util/ledger_util.go

```diff
@@ -64,6 +65,10 @@ func createInitState(genesis *bookkeeping.Genesis) (ledgercore.InitState, error)
 // MakeLedger opens a ledger, initializing if necessary.
 func MakeLedger(logger *log.Logger, inMemory bool, genesis *bookkeeping.Genesis, dataDir string) (*ledger.Ledger, error) {
 	const prefix = "ledger"
+	err := os.MkdirAll(dataDir, 0755)
+	if err != nil {
+		return nil, fmt.Errorf("MakeProcessor() failed to create '%s': %w", dataDir, err)
+	}
```

Comment:
Automatically create the local ledger directory if it's missing. While I was testing different configurations, this kept coming up.